packages/plugin-autocapture-browser/src/data-extractor.ts

```diff
@@ -192,32 +201,27 @@ export class DataExtractor {
     return undefined;
   };
 
-  combineText = (element: Element): string => {
-    let text = '';
-    if (isNonSensitiveElement(element) && element.childNodes && element.childNodes.length) {
-      element.childNodes.forEach((child) => {
-        let childText = '';
-        if (isTextNode(child)) {
-          if (child.textContent) {
-            childText = child.textContent;
-          }
-        } else {
-          childText = this.combineText(child as Element);
-        }
-        text += childText
-          .split(/(\s+)/)
-          .filter(this.isNonSensitiveString)
-          .join('')
-          .replace(/[\r\n]/g, ' ')
-          .replace(/[ ]+/g, ' ')
-          .substring(0, 255);
+  getText = (element: Element): string => {
+    // Check if element or any parent has data-amp-mask attribute
+    const hasMaskAttribute = element.closest('[' + constants.TEXT_MASK_ATTRIBUTE + ']') !== null;
+    if (hasMaskAttribute) {
+      return constants.MASKED_TEXT_VALUE;
+    }
+    let output = '';
+    if (!element.querySelector(`[${constants.TEXT_MASK_ATTRIBUTE}], [contenteditable]`)) {
+      output = (element as HTMLElement).innerText || (element as HTMLElement).textContent || '';
```

Comment:
remove `|| (element as HTMLElement).textContent ` unless there's a specific scenario you think that requires this.

Reply:
done